rust/src/tests/mknod.rs

```diff
@@ -49,3 +51,25 @@ fn changed_time_fields_success(ctx: &mut TestContext) {
             mknod_wrapper(&path, Mode::from_bits_truncate(0o644)).unwrap();
         });
 }
+
+// mknod/01.t
+assert_enotdir_comp!(mknod(~path, SFlag::S_IFIFO, Mode::empty(), 0));
+
+crate::test_case! {
```

Comment:
Is the only reason that this can't use the macro the fact that it needs root?

Reply:
No, should I add a macro form for this?

Reply:
What else does it need?

Reply:
Sorry, it doesn't need anything besides the root privileges.

Reply:
Well, it's up to you.  If you can save code by moving some logic into the macro, then I suggest you do.  But if this is the only place where you would need it, then it might not be worthwhile.